pkg/glob/glob.go

```diff
@@ -52,8 +53,7 @@ func (p Pattern) Glob(cb func(PathInfo) bool) bool {
 }
 
 func isDrive(s string) bool {
-	return len(s) == 2 && s[1] == ':' &&
-		(('a' <= s[0] && s[1] <= 'z') || ('A' <= s[0] && s[0] <= 'Z'))
+	return len(s) == 2 && s[1] == ':' && unicode.IsLetter(rune(s[0]))
```

Comment:
`unicode.IsLetter` does much more than checking the rune is between A and Z. The behaviour *happens* to be correct since s[0] is a single byte, but I find this unnecessarily clever. A plain function to check ASCII letters is better:

```go
func isASCIILetter(b byte) bool
```

Reply:
I have no idea why you think this is clever since the unicode functions produce the expected results for ASCII bytes. If we're going to implement ASCII specific analogues it is probably better to bite the bullet and add a dependency on https://github.com/scott-ainsworth/go-ascii/. Yay or nay?

Reply:
I ran `grep "'0'" **.go` and found fewer instances (five?) of checking if a byte is an ASCII digit than I expected. Nonetheless, it illustrates that we should either depend on the `go-ascii` package for such tests or implement our own, equivalent, set of functions.

Reply:
One of our ASCII char tests is for an octal digit (see *pkg/parse/parse.go*). Which isn't supported explicitly by the `scott-aisnsworth/go-ascii` package. So that particular test would have to remain in its current form (`if r < '0' || r > '7' {`) unless abstracted into an elvish specific function. Argh! At this point I'm inclined to introduce package `ascii` (*pkg/ascii/ascii.go*) that provides the ASCII classifications required by Elvish. Then consistently use those functions rather than hand-coded equality tests like `r < '0' || r > '7'`.

Reply:
> I have no idea why you think this is clever since the unicode functions produce the expected results for ASCII bytes.

Clever code is when the writer tries to save a few keystrokes and incurs cognitive burden on the reader.

To understand that the version using unicode.IsLetter is working correctly, the reader needs to:

* Know that the Unicode definition of letters happen to include A-Z and a-z, and nothing more, in the ASCII range

* Read the code carefully and realize that it is converting a byte to a rune, so the rune can only be in the ASCII range

> If we're going to implement ASCII specific analogues it is probably better to bite the bullet and add a dependency on https://github.com/scott-ainsworth/go-ascii/. Yay or nay?

No. A little copying is better than a little dependency.

Reply:
> Know that the Unicode definition of letters happen to include A-Z and a-z, and nothing more, in the ASCII range

I would expect anyone minimally competent with respect to the Go language and Unicode to know that fact. Even programmers looking at this code who don't satisfy that assumption will probably draw the obvious, correct, conclusion.  It's also no more clever than the existing logic. Consider, for example, that the existing code is incorrect if the encoding is EBCDIC. Yes, I know that isn't possible in practice but that isn't the point. There is nothing clever about using `unicode.IsLetter` other than the explicit type coercion from byte to rune. That, I will grant you, is clever in the sense you are using the term and it would be preferable to avoid it. So I'll introduce an `ascii` package to make such tests easier to read and less likely to be misunderstood by a human or the Go tool chain.

P.S., The unit tests which exercise `isDrive()` only pass uppercase `C:` or strings of length one. I learned this by using my un-merged trace package.

Reply:
> I would expect anyone minimally competent with respect to the Go language and Unicode to know that fact.

The [godoc for unicode.IsLetter](https://pkg.go.dev/unicode#IsLetter) reads:

> IsLetter reports whether the rune is a letter (category L).

If your definition of minimal competency with Unicode requires immediately knowing that the intersection between Unicode category L and ASCII is [a-zA-Z] without looking up how Unicode defines category L, I'd happily admit that I'm not minimally competent with Unicode.

My preferred definition of minimal competency with Unicode would be knowing enough Unicode to have a healthy fear of any Unicode concept - including seemingly banal concepts as "letter" in the Unicode context, and not using them in contexts that shouldn't require an understanding of Unicode.